### ISSUE TITLE
Difficulty Getting FDS6 to Run on a Mac

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#2234]
```
Please complete the following lines...

FDS Version: 6.1.2
SVN Revision Number:
Compile Date:
Smokeview Version/Revision: 6.1.12
Operating System:  Mac OSX 

Describe details of the issue below:

I've been working with FDS5 regularly until recently when I decided to start using
FDS6.  Today I downloaded the bundle for FDS6 for Mac OS X and have run into a problem
and can't seem to find a fix.

The downloaded program is "FDS_6.1.2-SMV_6.1.12_osx64.sh".  I followed the instructions
on the page entitled "Mac OS X Installation Notes" dated 9/18/14.  I then tried to
run an .fds input file by dragging it onto the launcher and received the following
message:


         FDS-SMV Launcher for OSX
         **** Version: 6.0.0 ****

         *** Input archive file: Halim1-FDS6.zip 

         *** Running Input File: Halim1-FDS6.fds


At that point, it appeared that any progress stopped.  I tried the same input file
on my Mac's FDS5 launcher and it works fine.

I next tried to run several of the FDS6 example problems but got the same result. 
I asked my son who uses FDS6 on a Windows machine to try my input file and it worked
fine in the Windows environment.

FYI, I'm using OSX version 10.7.5 on my MacBook Pro.  It has 8 GB of ram as well as
a 2.2 GHz Intel Core i7 processor.   As for the FDS version, I downloaded the file
"FDS_6.1.2-SMV_6.1.12_osx64" today so I think I have the latest one.  Might there be
a problem having both launchers for FDS5 for OS X and FDS6 for OS X both installed
on the same machine?

A message I saw on one of the Google sites indicated there may be a problem with memory
allocation on Macs and talked about how to adjust that.  I am not sure how the "sh,
bash" or other type commands work in order to allocate memory in the OS X system however.
 If that's a likely issue, are there any other suggestions on specific commands I should
try since I'm not familiar with Linux / Unix type syntax?

I'm attaching a copy of the input file to see if anything jumps out to you folks.

Thanks very much.

```

Original issue reported on code.google.com by `scarman58` on 2014-10-21 03:45:09

<hr>
- _Attachment: [Halim1-FDS6.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-2276/comment-0/Halim1-FDS6.fds)_

**Comment by gforney** [comment]
```
Steve:

I was able to run your case on my Macbook Pro (8 GB of RAM) running OS X 10.10 with
FDS 6.1.2 and the launcher included in the bundle. The case started to run and print
time step information after about 10 seconds.

The FDS 6 launcher explicitly points to the FDS 6 version. However, to rule out any
issues with the launcher, try to run the FDS case directly from the command line instead
of using the launcher:

1) To make sure that you are referencing the newest version of FDS, open the Terminal
app (similar to the command prompt on Windows), type 'fds', and press enter. You should
see FDS print the 6.1.2 version banner.

2) Run the FDS case in its own directory from the Terminal app using the 'fds' command.

It is possible that the launcher is hiding an FDS error that is occurring on your Mac
when the case starts up, or you need to wait longer for the case to start running,
which the above steps will help us determine.
```

Original issue reported on code.google.com by `koverholt` on 2014-10-21 04:02:27

**Comment by gforney** [comment]
```
I tried doing that with no luck.  Then I went into the Applications directory where
the bin files are located and ran the FDS executable file.  I received the following:

Last login: Tue Oct 21 07:37:14 on ttys000
Steves-MacBook-Pro:~ swcarman$ /Applications/FDS/FDS6/bin/fds ; exit;
Illegal instruction: 4
logout

[Process completed]

Does this look familiar?
```

Original issue reported on code.google.com by `scarman58` on 2014-10-21 14:46:44

**Comment by gforney** [comment]
```
Can you type the following two commands in the Terminal app and tell me what the output
is?

getconf LONG_BIT

uname -a
```

Original issue reported on code.google.com by `koverholt` on 2014-10-21 15:19:51

**Comment by gforney** [comment]
```
Last login: Tue Oct 21 07:45:21 on ttys001
Steves-MacBook-Pro:~ swcarman$ getconf LONG_BIT
64
Steves-MacBook-Pro:~ swcarman$ uname -a
Darwin Steves-MacBook-Pro.local 11.4.2 Darwin Kernel Version 11.4.2: Thu Aug 23 16:25:48
PDT 2012; root:xnu-1699.32.7~1/RELEASE_X86_64 x86_64
Steves-MacBook-Pro:~ swcarman$ 

 One question I had is whether there's an inherent problem having both FDS5 and FDS6
running on the Mac, both residing in the Applications folder.  I wondered if there
might be a path violation or confusion between the two files.
```

Original issue reported on code.google.com by `scarman58` on 2014-10-21 15:22:00

**Comment by gforney** [comment]
```
Thanks, that confirms you are running in 64-bit mode, which is good.

I am almost certain that your error is happening because FDS 6 is not compatible with
your older version of OS X. We compile FDS on the latest NIST-approved version of OS
X, which is currently 10.9, although 10.10 was released last week and we will be upgrading
as soon as we can. There are compiler options that we can use to force backwards compatibility
of OS X, but we've had stability and accuracy issues when we've used these before.

It sounds like updating your version of OS X will fix your problem. The last two versions
of OS X were released as free upgrades.
```

Original issue reported on code.google.com by `koverholt` on 2014-10-21 15:30:25

**Comment by gforney** [comment]
```
And to answer your other question, there is no problem with running FDS 5 and FDS 6
on the same machine. Our installers are set up to link the shortcut "fds" to the most
recent installed version of FDS.
```

Original issue reported on code.google.com by `koverholt` on 2014-10-21 15:31:48

**Comment by gforney** [comment]
```
Well I have this same case running at this time on the same Mac using FDS5 also using
an installer icon on the dock.
```

Original issue reported on code.google.com by `scarman58` on 2014-10-21 15:37:14

**Comment by gforney** [comment]
```
Thanks Kris.  I'll give that update of OS X a try.
```

Original issue reported on code.google.com by `scarman58` on 2014-10-21 15:39:02

**Comment by gforney** [comment]
```
With FDS 5, we compiled on whatever version of OS X was available to us 4 years ago,
which is why you are only seeing this error with FDS 6. We make an effort to always
compile with the latest versions of the various operating systems.

I'll close issue this for now, but post here again if updating OS X does not solve
the problem.
```

Original issue reported on code.google.com by `koverholt` on 2014-10-21 15:44:45

**Comment by gforney** [comment]
```
The OP reported to me via email that updating to the latest version of OS X 10.10 (Yosemite)
fixed the problem, and FDS 6.1.2 is now working on his Mac machine. Thanks for letting
us know the solution.
```

Original issue reported on code.google.com by `koverholt` on 2014-10-22 01:59:21
